packages/dependency-extraction-webpack-plugin/test/build.js

```diff
@@ -46,14 +46,21 @@ describe.each( configFixtures )( 'Webpack `%s`', ( configCase ) => {
 			webpack( options, ( err, stats ) => {
 				expect( err ).toBeNull();
 
-				const assetFiles = glob(
-					`${ outputDirectory }/*.asset.@(json|php)`
+				let assetFiles = glob(
+					`${ outputDirectory }/assets.@(json|php)`
 				);
-				const expectedLength =
-					typeof options.entry === 'object'
-						? Object.keys( options.entry ).length
-						: 1;
-				expect( assetFiles ).toHaveLength( expectedLength );
+				if ( assetFiles.length > 0 ) {
+					expect( assetFiles ).toHaveLength( 1 );
+				} else {
+					assetFiles = glob(
```

Comment:
Rather than making a special case, it would be nice to make the glob match the combined assets file and adjus expected length. The glob shouldn't be too bad, and we ought to be able to detect the combined assets option with something like the following:

```js
const hasCombinedAssets = ( options.plugins || [] ).some( plugin => ( plugin.options || {} ).combineAssets;
const entrypointCount = typeof options.entry === 'object' ? Object.keys( options.entry ).length : 1
const expectedLength = hasCombinedAssets ? 1 : entrypointCount;
```

Untested, but that would let us remove some branching from these already confusing tests.

Reply:
I didn't know how to access plugins, if that works then it's much better :)

I even had an updated glob call that was able to match this new file but without this check, it was cumbersome to handle.

Reply:
e81e2a602e45e9af47b1fd70c615c4aef21997d4 works like a charm, thanks again for raising it 👍